core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -598,6 +598,59 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+  test("run with ShuffleMapStage retry") {
+    val firstRDD = new MyRDD(sc, 3, Nil)
+    val firstShuffleDep = new ShuffleDependency(firstRDD, null)
+    val firstShuffleId = firstShuffleDep.shuffleId
+    val shuffleMapRdd = new MyRDD(sc, 3, List(firstShuffleDep))
+    val shuffleDep = new ShuffleDependency(shuffleMapRdd, null)
+    val reduceRdd = new MyRDD(sc, 1, List(shuffleDep))
+    submit(reduceRdd, Array(0))
+
+    complete(taskSets(0), Seq(
+      (Success, makeMapStatus("hostB", shuffleMapRdd.partitions.size)),
+      (Success, makeMapStatus("hostB", shuffleMapRdd.partitions.size)),
+      (Success, makeMapStatus("hostA", shuffleMapRdd.partitions.size))
+    ))
+
+    complete(taskSets(1), Seq(
+      (Success, makeMapStatus("hostA", reduceRdd.partitions.size)),
+      (Success, makeMapStatus("hostB", reduceRdd.partitions.size))
+    ))
+    runEvent(ExecutorLost("exec-hostA"))
+    // Resubmit already succcessd in hostA task
+    runEvent(CompletionEvent(taskSets(1).tasks(0), Resubmitted,
+      null, null, createFakeTaskInfo(), null))
+
+    // Cause mapOutTracker remove hostA outputs for taskset(0).
+    // Task that resubmitted will fetch matadata failed.
+    runEvent(CompletionEvent(taskSets(1).tasks(0),
+      FetchFailed(null, firstShuffleId, 2, 0, "Fetch failed"),
+      null, null, createFakeTaskInfo(), null))
+
+    // FetchFailed cause resubmit failed Stages.
+    scheduler.resubmitFailedStages()
+
+    runEvent(CompletionEvent(taskSets(0).tasks(2), Success,
+      makeMapStatus("hostC", shuffleMapRdd.partitions.size), null, createFakeTaskInfo(), null))
```

Comment:
If this is for the resubmit of stage 0, then I think you really want `taskSets(2).tasks(0)`.  In fact to be really sure its the right event, you could do something like

```scala
    val stage0Resubmit = taskSets(2)
    assert(stage0Resubmit.stageId == 0)
    assert(stage0Resubmit.attempt === 1)
    val task = stage0Resubmit.tasks(0)
    assert(task.partitionId === 2)
    runEvent(CompletionEvent(task, Success,
      makeMapStatus("hostC", shuffleMapRdd.partitions.size), null, createFakeTaskInfo(), null))
```